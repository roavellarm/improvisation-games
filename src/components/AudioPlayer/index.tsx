import React from 'react'
import ReactAudioPlayer from 'react-audio-player'
import styled from 'styled-components'

const StyledPlayer = styled<any>(ReactAudioPlayer)`
  position: relative;
  width: 100%;
  margin-top: 1rem;
  outline: none;
  z-index: 1 !important;
`

type AudioPlayerProps = {
  title: string
  src: string | undefined
}

export default function AudioPlayer({ title, src }: AudioPlayerProps) {
  return <StyledPlayer controls title={`Exemplo ${title}`} src={src} />
}
