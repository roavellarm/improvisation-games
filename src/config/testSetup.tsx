import React, { ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'

import theme from 'styles/theme'

export const withTheme = (Component: ReactNode) => (
  <ThemeProvider theme={theme}>{Component}</ThemeProvider>
)
