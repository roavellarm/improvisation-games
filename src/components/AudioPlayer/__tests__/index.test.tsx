import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import { withTheme } from 'config/testSetup'
import AudioPlayer from '..'

describe('Audio Player', () => {
  it('should render correctly', () => {
    const tree = render(withTheme(<AudioPlayer title="Foo bar" src="audioSource" />))
    expect(tree).toMatchSnapshot()
  })
})
