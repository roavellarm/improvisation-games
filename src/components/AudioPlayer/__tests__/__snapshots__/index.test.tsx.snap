// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio Player should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <audio
        class="react-audio-player sc-bdfBwQ juLNas"
        controls=""
        id=""
        preload="metadata"
        src="audioSource"
        title="Exemplo Foo bar"
      >
        <p>
          Your browser does not support the 
          <code>
            audio
          </code>
           element.
        </p>
      </audio>
    </div>
  </body>,
  "container": <div>
    <audio
      class="react-audio-player sc-bdfBwQ juLNas"
      controls=""
      id=""
      preload="metadata"
      src="audioSource"
      title="Exemplo Foo bar"
    >
      <p>
        Your browser does not support the 
        <code>
          audio
        </code>
         element.
      </p>
    </audio>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
